docs/troubleshooting-guide/performance.md

```diff
@@ -0,0 +1,110 @@
+## Performance
+
+In order to troubleshoot performance problems, you first need to understand what happens when a question or dashboard is updated in Metabase:
+
+1. Your browser goes to a web page that shows a Metabase question or dashboard.
+
+2. The JavaScript in our front end sends a request to our back end. (It also does this as you type in filter controls or search fields to fill ahead.)
+
+3. Our back end checks its cache to see if it can re-use a recently-fetched result rather than sending another request to the database. If so, that result comes back right away.
+
+4. Otherwise, if the question was created using the notebook editor, we translate the MBQL representation of the question into SQL. Storing an abstract representation of the question and translating it on the fly lets us create SQL specific to the database you're using: while SQL is supposed to be a standard, its various implementations differ in many ways.
+
+5. The back end then sends the request to the database. It needs a connection to the database in order to do this; like most applications, Metabase maintains a pool of connections that it re-uses rather than making a fresh connection each time.
+
+6. The database executes the query and returns the result to the back end.
+
+7. The back end forwards those results to the front end.
+
+8. The front end creates the HTML to display them.
+
+## Specific Problems
+
+Each of the steps described is a potential performance bottleneck:
+
+### The Metabase instance is getting so much traffic that loading the HTML page is slow.
+
+This is pretty rare, since our front end is not very large and browsers cache our JavaScript, but it's easy to rule out.
+
+**How to detect this:** Nothing appears on the page (not even the controls), or elements appear one by one.
+
+**How to fix this:** Look at page load times in the browser console [FIXME: link]. If it's taking a long time to load our HTML, CSS, or JavaScript, check to see whether a proxy, firewall, or other network component is slowing things down.
+
+### Caching is disabled.
```

Comment:
Should we note that ad-hoc queries (unsaved questions) are not cached - nor is exports?

Reply:
incorporated - thank you.